commit ef5b31b920d63b8500b8533a0bbbd0cbde3f8c18
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Mon May 8 16:43:40 2017 +0100

    irq_bcm2836: Send event when onlining sleeping cores
    
    In order to reduce power consumption and bus traffic, it is sensible
    for secondary cores to enter a low-power idle state when waiting to
    be started. The wfe instruction causes a core to wait until an event
    or interrupt arrives before continuing to the next instruction.
    The sev instruction sends a wakeup event to the other cores, so call
    it from bcm2836_smp_boot_secondary, the function that wakes up the
    waiting cores during booting.
    
    It is harmless to use this patch without the corresponding change
    adding wfe to the ARMv7/ARMv8-32 stubs, but if the stubs are updated
    and this patch is not applied then the other cores will sleep forever.
    
    See: https://github.com/raspberrypi/linux/issues/1989
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/irqchip/irq-bcm2836.c b/drivers/irqchip/irq-bcm2836.c
index c4e151451cf8..bee4d2d2ebac 100644
--- a/drivers/irqchip/irq-bcm2836.c
+++ b/drivers/irqchip/irq-bcm2836.c
@@ -227,6 +227,9 @@ static int __init bcm2836_smp_boot_secondary(unsigned int cpu,
 	writel(secondary_startup_phys,
 	       intc.base + LOCAL_MAILBOX3_SET0 + 16 * cpu);
 
+	dsb(sy); /* Ensure write has completed before waking the other CPUs */
+	sev();
+
 	return 0;
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1008_ef5b31b920d63b8500b8533a0bbbd0cbde3f8c18.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1008_ef5b31b920d63b8500b8533a0bbbd0cbde3f8c18.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
